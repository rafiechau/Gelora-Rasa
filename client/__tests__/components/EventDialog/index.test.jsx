import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import EventDialog from '@components/EventDialog';

// Mock module yang tidak perlu
jest.mock('react-redux', () => ({
  useDispatch: () => jest.fn(),
  connect: () => (Component) => (props) => <Component {...props} />,
}));

jest.mock('react-router-dom', () => ({
  useNavigate: () => jest.fn(),
}));

jest.mock('react-intl', () => ({
  FormattedMessage: jest.fn(({ id }) => <span>{id}</span>),
  injectIntl: (Component) => (props) => <Component {...props} intl={{ formatMessage: (msg) => msg.id }} />,
}));

describe('EventDialog', () => {
  const mockOnClose = jest.fn();
  const mockEvent = {
    id: 1,
    eventName: 'Konser Musik Akbar',
    date: '2023-04-01',
    time: '19:00',
    registrationDealine: '2023-03-30',
    address: 'Jalan Merdeka No. 123',
    venueName: 'Stadion Merdeka',
    price: 50000,
    stok: 100,
    type: 'offline',
    status: 'active',
    description: '<p>Konser musik tahunan dengan bintang tamu terkenal.</p>',
    locationId: 1,
    categoryId: 1,
    image: '/path/to/image.jpg',
    Location: {
      namaProvinsi: 'Jakarta',
    },
    Category: {
      categoryName: 'Musik',
    },
  };
  const mockLocations = [{ id: 1, namaProvinsi: 'Jakarta' }];
  const mockCategories = [{ id: 1, categoryName: 'Music' }];
  const mockToken = 'fakeToken';
  const mockUser = { id: 1, role: 'user' };

  beforeEach(() => {
    render(
      <EventDialog
        open
        onClose={mockOnClose}
        mode="edit"
        myEvent={mockEvent}
        locations={mockLocations}
        categories={mockCategories}
        token={mockToken}
        user={mockUser}
        intl={{ formatMessage: (msg) => msg.id }}
      />
    );
  });

  test('Should render correctly', () => {
    expect(screen.getByTestId('event-dialog')).toBeInTheDocument();
  });

  test('Should call onClose when close button is clicked', () => {
    fireEvent.click(screen.getByTestId('close-button'));
    expect(mockOnClose).toHaveBeenCalled();
  });

  test('Should call onSubmit when submit button is clicked', () => {
    fireEvent.click(screen.getByTestId('submit-button'));
  });

  test('Should match snapshot', () => {
    const dialog = screen.getByTestId('event-dialog');
    expect(dialog).toMatchSnapshot();
  });
});
