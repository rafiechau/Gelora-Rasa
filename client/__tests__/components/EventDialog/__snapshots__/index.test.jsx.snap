// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventDialog Should match snapshot 1`] = `
<div
  class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
  data-testid="event-dialog"
  role="presentation"
>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  />
  <div
    data-testid="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
    role="presentation"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    tabindex="-1"
  >
    <div
      aria-labelledby=":r3:"
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiDialog-paperFullWidth css-tlc64q-MuiPaper-root-MuiDialog-paper"
      role="dialog"
    >
      <h2
        class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
        id=":r3:"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          data-testid="close-button"
          style="position: absolute; right: 8px; top: 8px;"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CloseSharpIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </h2>
      <div
        class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
      >
        <form
          class="MuiBox-root css-h4tqr2"
          novalidate=""
        >
          <div
            class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
          >
            <label
              class="labelInput"
              for="eventName"
            >
              app_event_name
              <input
                class="input undefined"
                id="eventName"
                name="eventName"
                type="text"
                value=""
              />
            </label>
          </div>
          <div
            class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
          >
            <label
              class="labelInput"
              for="date"
            >
              app_event_date
              <input
                class="input undefined"
                id="date"
                name="date"
                type="date"
                value=""
              />
            </label>
          </div>
          <div
            class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
          >
            <label
              class="labelInput"
              for="time"
            >
              app_event_time_name
              <input
                class="input undefined"
                id="time"
                name="time"
                type="time"
                value=""
              />
            </label>
          </div>
          <div
            class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
          >
            <label
              class="labelInput"
              for="registrationDealine"
            >
              app_event_registration_dealine_name
              <input
                class="input undefined"
                id="registrationDealine"
                name="registrationDealine"
                type="date"
                value=""
              />
            </label>
          </div>
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              id="demo-simple-select-label"
            >
              <span>
                app_event_type_name
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="demo-simple-select-label demo-simple-select"
                class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="demo-simple-select"
                role="button"
                tabindex="0"
              >
                <span>
                  app_offline
                </span>
              </div>
              <input
                aria-hidden="true"
                aria-invalid="false"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                name="type"
                tabindex="-1"
                value="offline"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-14lo706"
                >
                  <span>
                    Type Event
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
          >
            <label
              class="labelInput"
              for="address"
            >
              app_event_address_name
              <input
                class="input undefined"
                id="address"
                name="address"
                type="text"
                value=""
              />
            </label>
          </div>
          <div
            class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
          >
            <label
              class="labelInput"
              for="venueName"
            >
              app_event_venue_name
              <input
                class="input undefined"
                id="venueName"
                name="venueName"
                type="text"
                value=""
              />
            </label>
          </div>
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              id="demo-simple-select-label"
            >
              <span>
                app_event_status_name
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="demo-simple-select-label demo-simple-select"
                class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="demo-simple-select"
                role="button"
                tabindex="0"
              >
                <span>
                  app_active
                </span>
              </div>
              <input
                aria-hidden="true"
                aria-invalid="false"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                name="status"
                tabindex="-1"
                value="active"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-14lo706"
                >
                  <span>
                    Status Event
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <label
            class="dropContainer"
            for="images"
          >
            <img
              alt="Preview"
              class="imagePreview"
              src="http://localhost:5000//path/to/image.jpg"
            />
            <input
              accept="image/*"
              id="images"
              style="display: none;"
              type="file"
            />
          </label>
          <div
            class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
          >
            <label
              class="labelInput"
              for="price"
            >
              app_event_price_name
              <input
                class="input undefined"
                id="price"
                name="price"
                type="number"
                value=""
              />
            </label>
          </div>
          <div
            class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
          >
            <label
              class="labelInput"
              for="stok"
            >
              app_event_stok_name
              <input
                class="input undefined"
                id="stok"
                name="stok"
                type="number"
                value=""
              />
            </label>
          </div>
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              id="location-select-label"
            >
              <span>
                app_event_location
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="location-select-label location-select"
                class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="location-select"
                role="button"
                tabindex="0"
              >
                Jakarta
              </div>
              <input
                aria-hidden="true"
                aria-invalid="false"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                name="locationId"
                tabindex="-1"
                value="1"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-14lo706"
                >
                  <span>
                    Location
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              id="category-select-label"
            >
              <span>
                app_event_category
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="category-select-label category-select"
                class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="category-select"
                role="button"
                tabindex="0"
              >
                Music
              </div>
              <input
                aria-hidden="true"
                aria-invalid="false"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                name="categoryId"
                tabindex="-1"
                value="1"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-14lo706"
                >
                  <span>
                    Category
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="quill "
          >
            <div
              class="ql-toolbar ql-snow"
            >
              <span
                class="ql-formats"
              >
                <span
                  class="ql-header ql-picker"
                >
                  <span
                    aria-controls="ql-picker-options-3"
                    aria-expanded="false"
                    class="ql-picker-label"
                    role="button"
                    tabindex="0"
                  >
                    <svg
                      viewBox="0 0 18 18"
                    >
                       
                      <polygon
                        class="ql-stroke"
                        points="7 11 9 13 11 11 7 11"
                      />
                       
                      <polygon
                        class="ql-stroke"
                        points="7 7 9 5 11 7 7 7"
                      />
                       
                    </svg>
                  </span>
                  <span
                    aria-hidden="true"
                    class="ql-picker-options"
                    id="ql-picker-options-3"
                    tabindex="-1"
                  >
                    <span
                      class="ql-picker-item"
                      data-value="1"
                      role="button"
                      tabindex="0"
                    />
                    <span
                      class="ql-picker-item"
                      data-value="2"
                      role="button"
                      tabindex="0"
                    />
                    <span
                      class="ql-picker-item"
                      data-value="3"
                      role="button"
                      tabindex="0"
                    />
                    <span
                      class="ql-picker-item"
                      role="button"
                      tabindex="0"
                    />
                  </span>
                </span>
                <select
                  class="ql-header"
                  style="display: none;"
                >
                  <option
                    value="1"
                  />
                  <option
                    value="2"
                  />
                  <option
                    value="3"
                  />
                  <option
                    selected="selected"
                  />
                </select>
              </span>
              <span
                class="ql-formats"
              >
                <button
                  class="ql-bold"
                  type="button"
                >
                  <svg
                    viewBox="0 0 18 18"
                  >
                     
                    <path
                      class="ql-stroke"
                      d="M5,4H9.5A2.5,2.5,0,0,1,12,6.5v0A2.5,2.5,0,0,1,9.5,9H5A0,0,0,0,1,5,9V4A0,0,0,0,1,5,4Z"
                    />
                     
                    <path
                      class="ql-stroke"
                      d="M5,9h5.5A2.5,2.5,0,0,1,13,11.5v0A2.5,2.5,0,0,1,10.5,14H5a0,0,0,0,1,0,0V9A0,0,0,0,1,5,9Z"
                    />
                     
                  </svg>
                </button>
                <button
                  class="ql-italic"
                  type="button"
                >
                  <svg
                    viewBox="0 0 18 18"
                  >
                     
                    <line
                      class="ql-stroke"
                      x1="7"
                      x2="13"
                      y1="4"
                      y2="4"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="5"
                      x2="11"
                      y1="14"
                      y2="14"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="8"
                      x2="10"
                      y1="14"
                      y2="4"
                    />
                     
                  </svg>
                </button>
                <button
                  class="ql-underline"
                  type="button"
                >
                  <svg
                    viewBox="0 0 18 18"
                  >
                     
                    <path
                      class="ql-stroke"
                      d="M5,3V9a4.012,4.012,0,0,0,4,4H9a4.012,4.012,0,0,0,4-4V3"
                    />
                     
                    <rect
                      class="ql-fill"
                      height="1"
                      rx="0.5"
                      ry="0.5"
                      width="12"
                      x="3"
                      y="15"
                    />
                     
                  </svg>
                </button>
                <button
                  class="ql-link"
                  type="button"
                >
                  <svg
                    viewBox="0 0 18 18"
                  >
                     
                    <line
                      class="ql-stroke"
                      x1="7"
                      x2="11"
                      y1="7"
                      y2="11"
                    />
                     
                    <path
                      class="ql-even ql-stroke"
                      d="M8.9,4.577a3.476,3.476,0,0,1,.36,4.679A3.476,3.476,0,0,1,4.577,8.9C3.185,7.5,2.035,6.4,4.217,4.217S7.5,3.185,8.9,4.577Z"
                    />
                     
                    <path
                      class="ql-even ql-stroke"
                      d="M13.423,9.1a3.476,3.476,0,0,0-4.679-.36,3.476,3.476,0,0,0,.36,4.679c1.392,1.392,2.5,2.542,4.679.36S14.815,10.5,13.423,9.1Z"
                    />
                     
                  </svg>
                </button>
              </span>
              <span
                class="ql-formats"
              >
                <button
                  class="ql-list"
                  type="button"
                  value="ordered"
                >
                  <svg
                    viewBox="0 0 18 18"
                  >
                     
                    <line
                      class="ql-stroke"
                      x1="7"
                      x2="15"
                      y1="4"
                      y2="4"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="7"
                      x2="15"
                      y1="9"
                      y2="9"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="7"
                      x2="15"
                      y1="14"
                      y2="14"
                    />
                     
                    <line
                      class="ql-stroke ql-thin"
                      x1="2.5"
                      x2="4.5"
                      y1="5.5"
                      y2="5.5"
                    />
                     
                    <path
                      class="ql-fill"
                      d="M3.5,6A0.5,0.5,0,0,1,3,5.5V3.085l-0.276.138A0.5,0.5,0,0,1,2.053,3c-0.124-.247-0.023-0.324.224-0.447l1-.5A0.5,0.5,0,0,1,4,2.5v3A0.5,0.5,0,0,1,3.5,6Z"
                    />
                     
                    <path
                      class="ql-stroke ql-thin"
                      d="M4.5,10.5h-2c0-.234,1.85-1.076,1.85-2.234A0.959,0.959,0,0,0,2.5,8.156"
                    />
                     
                    <path
                      class="ql-stroke ql-thin"
                      d="M2.5,14.846a0.959,0.959,0,0,0,1.85-.109A0.7,0.7,0,0,0,3.75,14a0.688,0.688,0,0,0,.6-0.736,0.959,0.959,0,0,0-1.85-.109"
                    />
                     
                  </svg>
                </button>
                <button
                  class="ql-list"
                  type="button"
                  value="bullet"
                >
                  <svg
                    viewBox="0 0 18 18"
                  >
                     
                    <line
                      class="ql-stroke"
                      x1="6"
                      x2="15"
                      y1="4"
                      y2="4"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="6"
                      x2="15"
                      y1="9"
                      y2="9"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="6"
                      x2="15"
                      y1="14"
                      y2="14"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="3"
                      x2="3"
                      y1="4"
                      y2="4"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="3"
                      x2="3"
                      y1="9"
                      y2="9"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="3"
                      x2="3"
                      y1="14"
                      y2="14"
                    />
                     
                  </svg>
                </button>
              </span>
              <span
                class="ql-formats"
              >
                <button
                  class="ql-clean"
                  type="button"
                >
                  <svg
                    class=""
                    viewBox="0 0 18 18"
                  >
                     
                    <line
                      class="ql-stroke"
                      x1="5"
                      x2="13"
                      y1="3"
                      y2="3"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="6"
                      x2="9.35"
                      y1="12"
                      y2="3"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="11"
                      x2="15"
                      y1="11"
                      y2="15"
                    />
                     
                    <line
                      class="ql-stroke"
                      x1="15"
                      x2="11"
                      y1="11"
                      y2="15"
                    />
                     
                    <rect
                      class="ql-fill"
                      height="1"
                      rx="0.5"
                      ry="0.5"
                      width="7"
                      x="2"
                      y="14"
                    />
                     
                  </svg>
                </button>
              </span>
            </div>
            <div
              class="ql-container ql-snow"
            >
              <div
                class="ql-editor"
                contenteditable="true"
                data-gramm="false"
              >
                <p>
                  Konser musik tahunan dengan bintang tamu terkenal.
                </p>
              </div>
              <div
                class="ql-clipboard"
                contenteditable="true"
                tabindex="-1"
              />
              <div
                class="ql-tooltip ql-hidden"
              >
                <a
                  class="ql-preview"
                  href="about:blank"
                  rel="noopener noreferrer"
                  target="_blank"
                />
                <input
                  data-formula="e=mc^2"
                  data-link="https://quilljs.com"
                  data-video="Embed URL"
                  type="text"
                />
                <a
                  class="ql-action"
                />
                <a
                  class="ql-remove"
                />
              </div>
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1vhaqj4-MuiButtonBase-root-MuiButton-root"
            data-testid="submit-button"
            tabindex="0"
            type="submit"
          >
            <span>
              app_update_event
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </form>
      </div>
    </div>
  </div>
  <div
    data-testid="sentinelEnd"
    tabindex="0"
  />
</div>
`;
